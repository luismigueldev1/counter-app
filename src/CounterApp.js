import React, { useState } from "react";
import PropTypes from "prop-types";

function CounterApp({ value = 0 }) {
  const [counter, setCounter] = useState(value);

  const handleAddCounter = () => {
    setCounter(counter + 1);
    //setCounter((prevState) => prevState + 1);
  };

  const handleSubCounter = () => setCounter(counter - 1);
  const handleReset = () => setCounter(value);
  return (
    <>
      <h1>CounterApp</h1>
      <p>{counter}</p>
      <button onClick={handleAddCounter}>+1</button>
      <button onClick={handleSubCounter}>-1</button>
      <button onClick={handleReset}>reset</button>
    </>
  );
}

CounterApp.propTypes = {
  value: PropTypes.number,
};

export default CounterApp;
