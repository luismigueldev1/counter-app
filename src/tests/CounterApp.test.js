import React from "react";
import { shallow } from "enzyme";
import CounterApp from "../CounterApp";

describe("Pruebas en <CounterApp>", () => {
  const value = 10;
  let wrapper = shallow(<CounterApp value={value} />);

  beforeEach(() => {
    wrapper = shallow(<CounterApp value={value} />);
  });

  test("Debe mostrar el componente correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe mostrar el valor por defect en 10", () => {
    const valueTest = 100;
    const wrapper = shallow(<CounterApp value={valueTest} />);

    const valorPorDefecto = Number(wrapper.find("p").text(value).trim());
    expect(valorPorDefecto).toBe(valueTest);
  });

  test("debe de incrementar el contador con el boton de +1 ", () => {
    wrapper.find("button").at(0).simulate("click");

    const counterText = Number(wrapper.find("p").text().trim());

    expect(counterText).toBe(value + 1);
  });

  test("debe de decrementar el contador con el boton de -1 ", () => {
    wrapper.find("button").at(1).simulate("click");
    const counterText = Number(wrapper.find("p").text().trim());
    expect(counterText).toBe(value - 1);
  });

  test("debe de resetear el valor por defecto con el boton reset ", () => {
    wrapper.find("button").at(0).simulate("click");
    wrapper.find("button").at(2).simulate("click");
    const counterText = Number(wrapper.find("p").text().trim());
    expect(counterText).toBe(value);
  });
});
