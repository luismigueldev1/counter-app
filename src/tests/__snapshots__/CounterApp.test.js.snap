// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <CounterApp> Debe mostrar el componente correctamente 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <p>
    10
  </p>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
  <button
    onClick={[Function]}
  >
    reset
  </button>
</Fragment>
`;
